MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -419,7 +419,7 @@ open class RouteStepProgress: NSObject {
     /**
      Returns an array of the calculated distances from the current intersection to the next intersection on the current step.
      */
-    @objc public var intersectionDistances = [CLLocationDistance]()
+    public var intersectionDistances: [CLLocationDistance]?
```

Comment:
Just curious, is it correct to remove the `@objc` declaration here?

Reply:
@akitchen I can't recall if the compiler complained. (PS: **I will be adding the @objc declaration back**,) 

Probably, might not apply to this...however, one problem with exposing the `CLLocationDistance` property is that you could store a non-CLLocationDistance-compatible value to it from Objective-C. This could be resolved with computed properties as we've done with other properties in the RouteProgress class. This variable is a stored property.

Reply:
I think it's a good point, but we can't stop all the ways developers can hurt themselves with Objective-C (or in general). Can't lightweight generics help serve at least as a hint in this regard on the ObjC side?

Reply:
@akitchen Yes, when I started my development in Swift 4.0, the `@objc` declaration wasn't happy with the optional array of the CLLocation. However, it's fixed in Swift 4.1.

Currently, our CI builds in Xcode 9.2, using the `@objc` here would trigger a failed build on bitrise.